retrofit/src/main/java/retrofit/Endpoint.java

```diff
@@ -8,7 +8,10 @@
 public abstract class Endpoint {
   /** Create an endpoint with the provided {@code url}. */
   public static Endpoint createFixed(String url) {
-    checkNotNull(url, "url == null");
+    checkNotNull(url, "URL == null");
+    if (url.trim().length() == 0) {
+      throw new IllegalArgumentException("URL is empty");
```

Comment:
@JakeWharton From aboe should this be lowercase as well or should I leave it as is?